hazelcast/src/test/java/com/hazelcast/test/starter/constructor/ConfigConstructor.java

```diff
@@ -69,12 +70,11 @@ private static Object cloneConfig(Object thisConfigObject, ClassLoader classload
         }
 
         Class thisConfigClass = thisConfigObject.getClass();
-        if (thisConfigClass.isPrimitive() || isJDKClass(thisConfigClass)) {
+        if (shouldProxy(thisConfigClass, new Class[0]) == RETURN_SAME) {
```

Comment:
In the future there will be an additional check in the `shouldProxy()` method, so it's the easiest to use it everywhere instead of making more `isXxx()` methods from the `HazelcastProxyFactory` public.